### DISCUSSION TITLE
Is it possible to pass in arguments as an object?

### DISCUSSION THREAD

**Discussion by jamauro** [wildhart/meteor.jobs#17] [Q&A]
It'd be nice if it didn't matter what order the arguments were passed in so that when using `Jobs.clear` later to match by argument, the argument's location didn't matter.

e.g. `Jobs.clear('pending', 'jobName', {foo: 'bar'})`

Is this already possible and I'm just missing something?

**Comment by wildhart** [comment]
I'm pretty sure that's already possible.  Have you tried?

However, when trying to clear the job with `Jobs.clear` if might only match if the parameter object is exactly the same because it doesn't use `$elemMatch``: https://github.com/wildhart/meteor.jobs/blob/4f2ce5e514f4c434c3d62eb3c75db8c7e5b341ba/jobs.ts#L344

**Reply by jamauro** [reply]
Right, it appears the parameter object has to be exactly the same when using `Jobs.clear`. It'd be nice if I could just match on some of the key/value pairs in the object.

So if I register a Job with `({userId, foo})`, I could later use `Jobs.clear('pending', 'jobName', {foo: 'bar'})` or `Jobs.clear('pending', 'jobName', {userId: '123'})`

**Reply by wildhart** [reply]
That could be possible, but would be a potential breaking change if some users are expecting the current behaviour.

Generally I would keep the `userId` by itself as the first parameter, then any additional options in an object as the 2nd parameter.  Then you can use `Jobs.clear('pending', 'jobName', userId)`.

Would that work for you?

**Reply by jamauro** [reply]
I can make it work as is but was hoping that I'd have a little more flexibility. :) Ideally, I'd like to be able to clear using 2 different unique identifiers. I get not wanting to introduce a breaking change. Maybe something to consider in the future if others find it useful.

**Reply by wildhart** [reply]
You can have all the flexibility you want by using [`Jobs.collection` ](https://github.com/wildhart/meteor.jobs#jobscollection)and implementing your own logic to delete particular jobs...
